### PR TITLE
Disable opt-sensitive tests under tailcall stress and balance pushd.

### PR DESCRIPTION
**Pull request by pgavlin**
Just what it says on the tin.

### CODE REVIEW

**Comment by pgavlin** [comment]
@RussKeldorph @jashook @dotnet/jit-contrib PTAL.

This fixes JIT/opt/perf/doublealign/locals failures that were occurring under tailcall stress.

File: tests/src/CLRTest.Execute.Bash.targets [hunk 1]
```diff
@@ -83,7 +83,7 @@ fi
       ]]></BashCLRTestEnvironmentCompatibilityCheck>
       <BashCLRTestEnvironmentCompatibilityCheck Condition="'$(JitOptimizationSensitive)' == 'true'"><![CDATA[
 $(BashCLRTestEnvironmentCompatibilityCheck)
-if [ \( ! -z "$COMPlus_JitStress" \) -o \( ! -z "$COMPlus_JitStressRegs" \) -o \( ! -z "$COMPlus_JITMinOpts" \) ]
+if [ \( ! -z "$COMPlus_JitStress" \) -o \( ! -z "$COMPlus_JitStressRegs" \) -o \( ! -z "$COMPlus_JITMinOpts" \) -o \( ! -z "$COMPlus_TailcallStress" \) ]
 then
   echo "SKIPPING EXECUTION BECAUSE ONE OR MORE OF (COMPlus_JitStress, COMPlus_JitStressRegs, COMPlus_JITMinOpts) IS SET"
```
Comment (RussKeldorph) [inline tests/src/CLRTest.Execute.Bash.targets]: >COMPlus_JitStress, COMPlus_JitStressRegs, COMPlus_JITMinOpts [](start = 51, length = 60)

Should update this message, too.

Comment (pgavlin) [inline tests/src/CLRTest.Execute.Bash.targets]: Ah, yes. Thanks.

File: tests/src/CLRTest.Execute.Batch.targets [hunk 2]
```diff
@@ -72,20 +72,23 @@ WARNING:   When setting properties based on their current state (for example:
...
+  popd
   Exit /b 0
 )
       ]]></BatchCLRTestEnvironmentCompatibilityCheck>
       <BatchCLRTestEnvironmentCompatibilityCheck Condition="'$(JitOptimizationSensitive)' == 'true'"><![CDATA[
 $(BatchCLRTestEnvironmentCompatibilityCheck)
-IF "%COMPlus_JitStress%"=="" IF "%COMPlus_JitStressRegs%"=="" IF "%COMPlus_JITMinOpts%"=="" goto :Compatible1
+IF "%COMPlus_JitStress%"=="" IF "%COMPlus_JitStressRegs%"=="" IF "%COMPlus_JITMinOpts%"=="" IF "%COMPlus_TailcallStress%"=="" goto :Compatible1
   ECHO SKIPPING EXECUTION BECAUSE ONE OR MORE OF (COMPlus_JitStress, COMPlus_JitStressRegs, COMPlus_JITMinOpts) IS SET
+  popd
```
Comment (RussKeldorph) [inline tests/src/CLRTest.Execute.Batch.targets]: Update message too.

File: tests/src/CLRTest.Execute.Batch.targets [hunk 3]
```diff
@@ -72,20 +72,23 @@ WARNING:   When setting properties based on their current state (for example:
 $(BatchCLRTestEnvironmentCompatibilityCheck)
 IF NOT "%COMPlus_GCStress%"=="" (
   ECHO SKIPPING EXECUTION BECAUSE COMPlus_GCStress IS SET
+  popd
```
Comment (RussKeldorph) [inline tests/src/CLRTest.Execute.Batch.targets]: Is [this](https://github.com/dotnet/coreclr/blob/36e988ee6593351f1e962118a99154df5eae0254/tests/src/CLRTest.Execute.Batch.targets#L347) the corresponding pushd?  I could have sworn the setlocal would preserve the working dir on exit, but if you're adding this I assume you've found that it doesn't.  If you're going to fix these, could you instead define exit blocks (possibly in _CLRTestExecutionScriptText) like:

Exit_success:
    popd
    Exit /b 0

Exit_1
    popd
    Exit /b 1

...

And then replace all the existing Exits with gotos?

**Comment by RussKeldorph** [comment]
:shipit:

File: tests/src/CLRTest.Execute.Batch.targets (hunk 3 above)
Comment (pgavlin) [inline tests/src/CLRTest.Execute.Batch.targets]: I considered that, but I think I'll leave it as-is for now.

Comment (pgavlin) [inline tests/src/CLRTest.Execute.Batch.targets]: (and yes, that is the corresponding pushd. I have found myself needing to repeatedly popd after running tests)

**Pull request merged**